### PR TITLE
Switch to tc-lib-monitor instead of tc-lib-stats

### PR DESCRIPTION
**Pull request by imbstack**
This should be pretty straightforward. There's a good bit less info being recorded than before I guess and this is kinda not DRY, but it seemed like the quickest way forward.

### CODE REVIEW

**Comment by jonasfj** [comment]
travis should be fixed...

**Comment by imbstack** [comment]
@jonasfj: haha, yeah. Didn't mean for you to see this back then. The tests are fixed now!

File: lib/client.js [hunk 1]
```diff
@@ -286,21 +268,10 @@ exports.createClient = function(reference, name) {
...
-            success:    success ? 1 : 0,
-            resolution: resolution,
-            target:     name,
-            baseUrl:    that._options.baseUrl
-          });
-        };
+      var monitor = null;
+      if (this._options.monitor) {
+        var start = process.hrtime();
+        monitor = this._options.monitor;
```
Comment (jonasfj) [inline lib/client.js]:
```js
var monitor = this._options.monitor;
if (monitor) {
  var start = process.hrtime();
}
```
or just:
```js
if (this._options.monitor) {
  var start = process.hrtime();
}
```

File: lib/client.js [hunk 2]
```diff
@@ -319,8 +290,10 @@ exports.createClient = function(reference, name) {
             // If request was successful, accept the result
             debug("Success calling: %s, (%s retries)",
                   entry.name, attempts - 1);
-            if (reportStats) {
-              reportStats(true, 'http-' + res.status);
+            if (monitor) {
+              var d = process.hrtime(start);
+              monitor.measure(entry.name, d[0] * 1000 + d[1] / 1000000);
```
Comment (jonasfj) [inline lib/client.js]: Can we do: `monitor.measure([entry.name, 'success'], ...` same for count...

File: lib/client.js [hunk 3]
```diff
@@ -353,8 +326,10 @@ exports.createClient = function(reference, name) {
               err.body = res.body;
               err.code = res.body.code || 'UnknownError';
               err.statusCode = res.status;
-              if (reportStats) {
-                reportStats(false, 'http-' + res.status);
+              if (monitor) {
+                var d = process.hrtime(start);
+                monitor.measure(entry.name, d[0] * 1000 + d[1] / 1000000);
+                monitor.count(entry.name);
```
Comment (jonasfj) [inline lib/client.js]: Can we do: `monitor.measure([entry.name, 'server-error'], ...` or `monitor.measure([entry.name, 'client-error'], ...`

depending on whether it's a 4xx or 5xx error...

File: lib/client.js [hunk 4]
```diff
@@ -367,12 +342,10 @@ exports.createClient = function(reference, name) {
...
-            if (reportStats) {
-              var errCode = err.code || err.name;
-              if (!errCode || typeof(errCode) !== 'string') {
-                errCode = 'Error';
-              }
-              reportStats(false, errCode);
+            if (monitor) {
+              var d = process.hrtime(start);
+              monitor.measure(entry.name, d[0] * 1000 + d[1] / 1000000);
+              monitor.count(entry.name);
```
Comment (jonasfj) [inline lib/client.js]: Can we do: `monitor.measure([entry.name, 'server-error'], ...` or `monitor.measure([entry.name, 'client-error'], ...`

depending on whether it's a 4xx or 5xx error...

File: lib/client.js [hunk 5]
```diff
@@ -129,27 +129,10 @@ var makeRequest = function(client, method, url, payload, query) {
  *   baseUrl:         'http://.../v1'   // baseUrl for API requests
  *   exchangePrefix:  'queue/v1/'       // exchangePrefix prefix
  *   retries:         5,                // Maximum number of retries
- *   stats:           function(obj) {}  // Callback for reporting statistics
+ *   monitor:         an instance of tc-lib-monitor.monitor({...})
```
Comment (jonasfj) [inline lib/client.js]:
```
 *   monitor:         await Monitor()   // From taskcluster-lib-monitor
```

**Comment by jonasfj** [comment]
I think having the distinction between: `client-error`, `server-error` and `success` is valuable...
But I don't think we need the `.all.` option, that is probably too much...

However, knowing if a services is getting a lot of client-errors or server-errors is very valuable as those are things we should potentially fix :)

**Comment by jonasfj** [comment]
r+, lovely

**Pull request merged**